### PR TITLE
use gopherwasm to wrap both js and wasm

### PR DESCRIPTION
**Pull request by xoviat**
vecty will support wasm soon. gopherwasm wraps both gopherjs and wasm compilers.

### CODE REVIEW

File: init.go [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-	"github.com/gopherjs/gopherjs/js"
+	"github.com/gopherjs/gopherwasm/js"
 )
 
 func init() {
-	js.Global.Set("onpopstate", func(e *js.Object) {
-		refreshRoutes()
-	})
+	js.Global().Set("onpopstate", js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		go refreshRoutes()
```
Comment (marwan-at-work) [inline init.go]: Out of curiosity, any reason why this is called in a goroutine? re-rendering is a synchronous process anyway and this is a callback so everything is already asynchronous.

File: redirect.go [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+
+	js.Global().Get("history").Call(
 		"pushState",
-		map[string]string{"redirectRoute": route},
+		map[string]interface{}{"redirectRoute": route},
 		route,
 		route,
 	)
-	refreshRoutes()
+	go refreshRoutes()
```
Comment (marwan-at-work) [inline redirect.go]: ditto

**Review by marwan-at-work — APPROVED** [review]
LGTM, just one question for curiosity and will merge. Thanks for the PR!

**Comment by xoviat** [comment]
The new goroutines were added when I was debugging a crash. I've removed them, and it works fine.

**Comment by marwan-at-work** [comment]
@xoviat thank you 👍

**Pull request merged**